alica_engine/include/engine/PlanBase.h

```diff
@@ -14,91 +14,93 @@
+    class AlicaEngine;
 
-	/**
-	 * A PlanBase holds the internal representation of the plan graph and issues all operations on it.
-	 * It is the most central object within the ALICA Engine.
-	 */
-	class PlanBase
-	{
-	public:
-		PlanBase(AlicaEngine* ae, Plan* masterplan);
-		~PlanBase();
-		condition_variable* getStepModeCV();
-		const shared_ptr<RunningPlan> getRootNode() const;
-		void setRootNode(shared_ptr<RunningPlan> rootNode);
-		void setRuleBook(RuleBook* ruleBook);
-		const ulong getloopInterval() const;
-		void setLoopInterval(ulong loopInterval);
-		void stop();
-		void start();
-		void addFastPathEvent(shared_ptr<RunningPlan> p);
-		shared_ptr<RunningPlan> getDeepestNode();
-		shared_ptr<RunningPlan> getRootNode();
-		Plan* getMasterPlan();
+    /**
+     * A PlanBase holds the internal representation of the plan graph and issues all operations on it.
+     * It is the most central object within the ALICA Engine.
+     */
+    class PlanBase
+    {
+    public:
+        PlanBase(AlicaEngine* ae, Plan* masterplan);
+        ~PlanBase();
+        std::condition_variable* getStepModeCV();
+        const std::shared_ptr<RunningPlan> getRootNode() const;
+        void setRootNode(std::shared_ptr<RunningPlan> rootNode);
+        const AlicaTime getloopInterval() const;
+        void setLoopInterval(ulong loopInterval);
+        void stop();
+        void start();
+        void addFastPathEvent(shared_ptr<RunningPlan> p);
+        std::shared_ptr<RunningPlan> getDeepestNode();
+        std::shared_ptr<RunningPlan> getRootNode();
 
-	private:
-		/**
-		 * List of RunningPlans scheduled for out-of-loop evaluation.
-		 */
-		queue<shared_ptr<RunningPlan>> fpEvents;
-		condition_variable* stepModeCV;
-//		void checkPlanBase(shared_ptr<RunningPlan> r);
+        /*const*/ Plan* getMasterPlan() const {return _masterPlan;}
+        bool isWaiting() const { return _isWaiting; }
+    private:
+        void run();
 
-	protected:
-		Plan* masterPlan;
-		shared_ptr<RunningPlan> rootNode;
-		shared_ptr<RunningPlan> deepestNode;
-		AlicaEngine* ae;
-		int treeDepth;
-		RuleBook* ruleBook;
-		TeamObserver* teamObserver;
-		IRoleAssignment* ra;
-		SyncModule* syncModel;
-		AuthorityManager* authModul;
-		IAlicaCommunication* statusPublisher;
-		IAlicaClock* alicaClock;
+        /**
+         * List of RunningPlans scheduled for out-of-loop evaluation.
+         */
 
-		AlicaTime loopTime;
-		AlicaTime lastSendTime;
-		AlicaTime minSendInterval;
-		AlicaTime maxSendInterval;
-		AlicaTime loopInterval;
-		AlicaTime lastSentStatusTime;
-		AlicaTime sendStatusInterval;
+        AlicaEngine* _ae;
+        Plan* _masterPlan;
+        
+        TeamObserver* _teamObserver;
+        IRoleAssignment* _ra;
+        SyncModule* _syncModel;
+        AuthorityManager* _authModul;
+        IAlicaCommunication* _statusPublisher;
+        IAlicaClock* _alicaClock;
 
-		bool running;
-		bool sendStatusMessages;
+        shared_ptr<RunningPlan> _rootNode;
```

Comment:
std::

Reply:
Yea, let's remove the using-statements one by one and use the compiler to guide us, that's simpler.